File: src/utils/stringUtils.ts
export function toKebabCase(str: string): string {
  return str
      .trim()
      .toLowerCase()
      .replace(/[^a-z0-9]+/g, '-') // Replace spaces and special characters with '-'
      .replace(/^-+|-+$/g, ''); // Remove dashes at the beginning or at the end
}

export function toCamelCase(str: string): string {
  return str
      .trim()
      .toLowerCase()
      .replace(/[^a-z0-9]+(.)/g, (_, chr) => chr.toUpperCase()) // Convert to camelCase
      .replace(/^(.)/, (_, chr) => chr.toLowerCase()); // Ensure the first letter is lowercase
}

export function toPascalCase(str: string): string {
  return str
      .trim()
      .toLowerCase()
      .replace(/[^a-z0-9]+(.)/g, (_, chr) => chr.toUpperCase()) // Convert to CamelCase
      .replace(/^(.)/, (_, chr) => chr.toUpperCase()); // Ensure the first letter is uppercase
}

export function toSnakeCase(str: string): string {
  return str
      .trim()
      .toLowerCase()
      .replace(/[^a-z0-9]+/g, '_') // Replace spaces and special characters with '_'
      .replace(/^_+|_+$/g, ''); // Remove underscores from the beginning or end
}

File: python/utils/string_utils.py
# utils/string_utils.py

import re

def to_kebab_case(s: str) -> str:
    return re.sub(r'^-+|-+$', '', re.sub(r'[^a-z0-9]+', '-', s.strip().lower()))

def to_camel_case(s: str) -> str:
    s = re.sub(r'[^a-z0-9]+(.)', lambda m: m.group(1).upper(), s.strip().lower())
    return s[0].lower() + s[1:] if s else s

def to_pascal_case(s: str) -> str:
    s = re.sub(r'[^a-z0-9]+(.)', lambda m: m.group(1).upper(), s.strip().lower())
    return s[0].upper() + s[1:] if s else s

def to_snake_case(s: str) -> str:
    return re.sub(r'^_+|_+$', '', re.sub(r'[^a-z0-9]+', '_', s.strip().lower()))
